data-collection/data-collection/View Controllers/Floating Panel View Controller/FloatingPanelController.swift

```diff
@@ -0,0 +1,792 @@
+            headerNavVC.view.widthAnchor.constraint(equalTo: headerNavVC.topViewController!.view.widthAnchor, multiplier: 1.0)
+        ])
+    }
+    
+    override func popViewController(animated: Bool) -> UIViewController? {
+        let returnVC = super.popViewController(animated: animated)
+        headerNavigationController?.popViewController(animated: animated)
+        
+        return returnVC
+    }
+}
+
+extension UIViewController {
+    /// Presents a `FloatingPanelController`.
+    /// - Parameter floatingPanel: The floating panel to display.
+    func presentFloatingPanel(_ floatingPanel: FloatingPanelController) {
+        addChild(floatingPanel)
+        view.addSubview(floatingPanel.view)
+        floatingPanel.didMove(toParent: self)
+        floatingPanel.view.translatesAutoresizingMaskIntoConstraints = false
+
+        // Set the constraints needed to position the floating panel on the
+        // left side of the screen, taking into account the size class insets.
+        if let floatingPanelView = floatingPanel.view {
+            floatingPanel.regularWidthConstraints = [
+                floatingPanelView.leadingAnchor.constraint(equalTo: view.safeAreaLayoutGuide.leadingAnchor, constant: floatingPanel.regularWidthInsets.left),
+                floatingPanelView.widthAnchor.constraint(equalToConstant: FloatingPanelController.floatingPanelWidth),
+                floatingPanelView.topAnchor.constraint(equalTo: view.safeAreaLayoutGuide.topAnchor, constant: floatingPanel.regularWidthInsets.top),
+                view.safeAreaLayoutGuide.bottomAnchor.constraint(greaterThanOrEqualTo: floatingPanelView.bottomAnchor, constant: floatingPanel.regularWidthInsets.bottom)
+            ]
+            
+            floatingPanel.compactWidthConstraints = [
+                floatingPanelView.leadingAnchor.constraint(equalTo: view.safeAreaLayoutGuide.leadingAnchor, constant: floatingPanel.compactWidthInsets.left),
+                floatingPanelView.trailingAnchor.constraint(equalTo: view.safeAreaLayoutGuide.trailingAnchor, constant: floatingPanel.compactWidthInsets.right),
+                floatingPanelView.bottomAnchor.constraint(equalTo: view.bottomAnchor, constant: floatingPanel.compactWidthInsets.bottom),
+                floatingPanelView.topAnchor.constraint(greaterThanOrEqualTo: view.safeAreaLayoutGuide.topAnchor, constant: floatingPanel.compactWidthInsets.top),
+                view.safeAreaLayoutGuide.bottomAnchor.constraint(greaterThanOrEqualTo: floatingPanel.headerView.bottomAnchor, constant: floatingPanel.compactWidthInsets.bottom)
+            ]
+        }
+    }
+    
+    /// Dismisses the given `FloatingPanelController`.
+    func dismissFloatingPanel(_ floatingPanel: FloatingPanelController) {
+        floatingPanel.willMove(toParent: nil)
+        floatingPanel.removeFromParent()
+        floatingPanel.view.removeFromSuperview()
```

Comment:
See [Implementing a Container View Controller](https://developer.apple.com/library/archive/featuredarticles/ViewControllerPGforiPhoneOS/ImplementingaContainerViewController.html).
```suggestion
        floatingPanel.willMove(toParent: nil)
        floatingPanel.view.removeFromSuperview()
        floatingPanel.removeFromParent()
```